src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/WebserviceController.php

```diff
@@ -66,17 +68,63 @@ public function indexAction(WebserviceKeyFilters $filters, Request $request)
 
         $configurationWarnings = $this->lookForWarnings();
 
+        $webserviceConfiguration = $this->get('prestashop.admin.webservice.form_data_provider')->getData();
+        $webserviceIsEnabled = $webserviceConfiguration['webservice_configuration']['enable_webservice'];
+        $webserviceStatus = [
+            'enabled' => $webserviceIsEnabled,
+            'isFunctional' => true,
+            'endpoint' => null,
+        ];
+
+        if ($webserviceIsEnabled) {
+            $expectedWebserviceEntryPoint = $request->getSchemeAndHttpHost() . self::WEBSERVICE_ENTRY_ENDPOINT;
+            $webServiceEntryPointIsReachable = $this->checkWebserviceEndpoint($expectedWebserviceEntryPoint);
+
+            $webserviceStatus['endpoint'] = $expectedWebserviceEntryPoint;
+            $webserviceStatus['isFunctional'] = $webServiceEntryPointIsReachable;
+        }
+
         return $this->render(
             '@PrestaShop/Admin/Configure/AdvancedParameters/Webservice/index.html.twig',
             [
                 'help_link' => $this->generateSidebarLink($request->get('_legacy_controller')),
                 'form' => $form->createView(),
                 'grid' => $presentedGrid,
                 'configurationWarnings' => $configurationWarnings,
+                'webserviceStatus' => $webserviceStatus,
             ]
         );
     }
 
+    /**
+     * @param string $url
+     *
+     * @return bool
+     */
+    private function checkWebserviceEndpoint($url): bool
+    {
+        if ($url === null) {
+            return false;
+        }
+
+        $ch = curl_init($url);
+        curl_setopt($ch, CURLOPT_TIMEOUT, 5);
```

Comment:
if you use curl, you need to allow configuration and customization. (Like for SSL) wdyt?

Reply:
Maybe but then it's a lot of config "just for a small check" 🤔

Reply:
If you're behind a custom proxy with SSL or a WAF, the request can fail because of missing configuration :/

Reply:
I know that I cannot handle _all_ configuration issues 😞 

So I put a "maybe" message: I dont say "endpoint does not work!" I say instead "it seems endpoint is not functional".

What I wanted to tackle is the usecase "URL rewriting is off so API does not work". I do not handle all usecases, but maybe we can merge this PR and slowly handle more and more usecases over time, when they are being reported as bug reports ?

See the screenshot https://github.com/PrestaShop/PrestaShop/pull/19335#issuecomment-648015337

Reply:
For a better usage, and customization, you should use a Guzzle client, a module will be able to change the configuration easily, wdyt?